packages/e2e-tests/config/setup-test-framework.js

```diff
@@ -190,13 +206,40 @@ async function runAxeTestsForBlockEditor() {
 	} );
 }
 
+/**
+ * Simulate slow network or throttled CPU if provided via environment variables.
+ */
+async function simulateAdverseConditions() {
+	if ( ! DOWNLOAD_THROUGHPUT && ! THROTTLE_CPU ) {
+		return;
+	}
+
+	const client = await page.target().createCDPSession();
+
+	if ( DOWNLOAD_THROUGHPUT ) {
+		// See: https://chromedevtools.github.io/devtools-protocol/tot/Network#method-emulateNetworkConditions
+		await client.send( 'Network.emulateNetworkConditions', {
+			// Simulated download speed (bytes/s)
+			downloadThroughput: Number( DOWNLOAD_THROUGHPUT ),
+		} );
+	}
+
+	if ( THROTTLE_CPU ) {
+		// See: https://chromedevtools.github.io/devtools-protocol/tot/Emulation#method-setCPUThrottlingRate
+		await client.send( 'Emulation.setCPUThrottlingRate', {
+			rate: Number( THROTTLE_CPU ),
+		} );
+	}
+}
+
 // Before every test suite run, delete all content created by the test. This ensures
 // other posts/comments/etc. aren't dirtying tests and tests don't depend on
 // each other's side-effects.
 beforeAll( async () => {
 	capturePageEventsForTearDown();
 	enablePageDialogAccept();
 	observeConsoleLogging();
+	simulateAdverseConditions();
```

Comment:
It's async so it probably should look as follows:

```suggestion
	await simulateAdverseConditions();
```

Reply:
> It's async so it probably should look as follows:

Good catch! Updated.